zrml/neo-swaps/src/types/liquidity_tree.rs

```diff
@@ -509,9 +515,18 @@ where
         if index == 0 { None } else { index.checked_sub(1)?.checked_div(2) }
     }
 
-    fn path_to_node(&self, index: u32) -> Result<Vec<u32>, DispatchError> {
+    fn path_to_node(
+        &self,
+        index: u32,
+        opt_iterations: Option<usize>,
+    ) -> Result<Vec<u32>, DispatchError> {
+        let remaining_iterations =
+            opt_iterations.unwrap_or(Self::max_depth().checked_add_res(&1)? as usize);
+        let remaining_iterations = remaining_iterations
+            .checked_sub(1)
+            .ok_or(LiquidityTreeError::MaxIterationsReached.into_dispatch_error::<T>())?;
```

Comment:
Thanks for being even defensive using the recursion approach :-) happy to see this.